include/triton/Dialect/TritonGPU/IR/TritonGPUOps.td

```diff
@@ -272,7 +273,8 @@ def TTG_MemDescReshapeOp : TTG_Op<"memdesc_reshape", [Pure,
     This doesn't affect the memory.
   }];
 
-  let arguments = (ins TTG_MemDescType:$src);
+  let arguments = (ins TTG_MemDescType:$src, DenseI64ArrayAttr:$shape);
```

Comment:
it's redundant to have the shape in the attribute as it will be available in the destination type.

Reply:
In general, and that's the point of this refactor, it's a bit tricky to infer the output as we also need to infer the allocShape. With this new API everything's handled in one place and it's done automatically for us, same as it happens in MemDescTransOp